### DISCUSSION TITLE
IsEquivalentTo()

### DISCUSSION THREAD

**Discussion by hischampion** [fluentassertions/fluentassertions#1965] [Ideas]
We end up writing stuff like this sometimes:
  ```cs
 public static bool IsEquivalentTo<T>(this T actual,
        T expected,
        Func<EquivalencyAssertionOptions<T>, EquivalencyAssertionOptions<T>>? config = null,
        string because = "")
{
    try
    {
        if (config == null)
        {
            actual.Should().BeEquivalentTo(expected, because);
        }
        else
        {
            actual.Should().BeEquivalentTo(expected, config, because);
        }
    }
    catch (Exception ex)
    {
        Debug.Write(ex.ToString());

        return false;
    }

    return true;
}
```

This would be for situations where you don't want to fail the test, but you want the nice comparison and output we all love from FluentAssertions. I would avoid doing this because of the flow control around the exception. It works, but it's not "kosher". 

Just wondering if a helper like this could be exposed?

Thanks for your time!

**Comment by dennisdoomen** [comment]
Just wrap the call to `BeEquivantTo` in a `using var scope = new AssertionScope` and just before disposing it, call `Discard` to see the errors without throwing an exception.
